cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
IMO this is our fault for allowing the inital sed change without a presubmit (https://github.com/kubernetes/kubernetes/issues/25031). It isn't specific to DNS but applies to all addons and static pods (so etcd, kube-proxy etc). What did we expect? :)